contributed/ra_dec_dc2_run2.2i_dask.ipynb

```diff
@@ -0,0 +1,1455 @@
+    "  * You can select the \"Running\" tab and then select the Notebook you want.\n",
+    "  * You could instead browse through the full filesystem path under the \"Files\" tab to find your Notebook, but that's a lot more clicking.  You may want to take this aproach to launch some other Notebook that's not currently running under JupyterHub.\n",
+    "\n",
+    "2. Requires:\n",
+    "```\n",
+    "healpy\n",
+    "holoviews\n",
+    "datashader\n",
+    "bokeh\n",
+    "pyarrow >= 0.13.1\n",
+    "```\n",
+    "\n",
+    "Up-to-date versions of each of these are available in `desc-python-bleed` kernel\n",
+    "\n",
+    "3. This was run using the `desc-python-bleed` kernel\n",
+    "\n",
+    "We directly use the DPDD Parquet files."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Import Needed Modules"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import math\n",
+    "import os\n",
+    "\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "\n",
+    "import astropy.units as u\n",
+    "import healpy as hp\n",
+    "\n",
+    "import scipy.interpolate"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import colorcet\n",
+    "\n",
+    "import dask\n",
+    "from dask.distributed import Client\n",
+    "\n",
+    "from bokeh.models import HoverTool\n",
+    "import dask.array as da\n",
+    "import dask.dataframe as dd\n",
+    "import datashader as ds\n",
+    "import holoviews as hv\n",
+    "from holoviews.operation import histogram\n",
+    "from holoviews.operation.datashader import datashade, shade, dynspread, rasterize\n",
+    "from holoviews.plotting.util import process_cmap\n",
+    "from holoviews.streams import RangeXY"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hv.extension('bokeh')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cmap = 'viridis'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Start our Dask Cluster\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We're only going to load the RA, Dec, so we don't need that much memory for the final product.\n",
+    "There's a 42 GB limit on memory directly in the JupyterHub environment, which we will suceed in staying under.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Start a local Dask Cluster"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "client = Client()"
```

Comment:
I suggest doing this to be explicit about how many cores and the amount of memory you are using.

```
from dask.distributed import LocalCluster

cluster = LocalCluster(n_workers=6,
threads_per_worker=1,
memory_limit='6Gb')
 
client = Client(cluster)
```

Note you then need to tell people how to access the dashboard.  

Related: you tell people to watch the 'Dask Dashboard when you zoom in and out.'  But never explicitly tell people to start it up etc.

Reply:
I actually don't know how to access the Dask Dashboard for  "Local" Dask cluster running on JupyterHub at NERSC.  The proxy service URL seems to only work for a scheduler running on a compute Node, not on the JupyterHub Node.

Do you know?

Reply:
Yes, I put in in my notes for using dask. It took me a bit to figure out but say you were on cori16 (check by starting a terminal). Then it would be

cori16-224.nersc.gov:8787

replace 16 which which ever machine you are on.

Reply:
also.. when you put just a client line there you will see something like this

<img width="559" alt="image" src="https://user-images.githubusercontent.com/5562740/122463980-4a0e0e80-cf84-11eb-8ff8-7f0f8b003d5f.png">

you are replacing the 127 part... if a bunch of people are using dask then the port might increment too.

Reply:
You don't want/need to use the proxy command thing in that case BTW.

Reply:
That URL gets me the link to the Notebook, but not the Dashboard.

Reply:
Specifically I am on `cori16`, the Dashboard was launched on 8787, but when I got to  https://cori16-224.nersc.gov:8787/status
I get (after a long pause) the Notebook I'm looking at in a new window.  But that only works once.  Future clicking hangs and never loads.

Reply:
Hmm... I just looked at it :)

I think maybe it is not https:  try just http:

Reply:
Woah... Yes `http://` works.  Thank you!  I would never have figured that out.

Reply:
Great.  FWIW, clue was the http://127.0.0.1 in the dashboard.